model/cells/zhong_cell.hpp

```diff
@@ -0,0 +1,199 @@
+//
+// Created by binybrion on 6/29/20.
+//
+
+#ifndef PANDEMIC_HOYA_2002_ZHONG_CELL_HPP
+#define PANDEMIC_HOYA_2002_ZHONG_CELL_HPP
+
+#include <cmath>
+#include <iostream>
+#include <vector>
+#include <cadmium/celldevs/cell/cell.hpp>
+#include <iomanip>
+#include "vicinity.hpp"
+#include "sir.hpp"
+#include "rates_configuration.h"
+
+using namespace std;
+using namespace cadmium::celldevs;
+
+template <typename T>
+class zhong_cell : public cell<T, std::string, sir, vicinity> {
+public:
+
+    template<typename X>
+    using cell_unordered = std::unordered_map<std::string, X>;
+
+    using cell<T, std::string, sir, vicinity>::simulation_clock;
+    using cell<T, std::string, sir, vicinity>::state;
+    using cell<T, std::string, sir, vicinity>::neighbors;
+
+    using config_type = rates_configurations;
+
+    using PhaseRates = std::vector<float>;
+
+    std::vector<PhaseRates> virulency_rates;
+    std::vector<PhaseRates> recovery_rates;
+    std::vector<PhaseRates> mobility_rates;
+
+    float disobedient = 0.0;
+    int precDivider = 1000;
+
+    zhong_cell() : cell<T, std::string, sir, vicinity>() {}
+
+    zhong_cell(std::string const &cell_id, cell_unordered<vicinity> const &neighborhood, sir initial_state, std::string const &delay_id, rates_configurations config) :
+    cell<T, std::string, sir, vicinity>(cell_id, neighborhood, initial_state, delay_id) {
+
+        virulency_rates = std::move(config.virulency_rates);
+        recovery_rates = std::move(config.recovery_rates);
+        mobility_rates = std::move(config.mobility_rates);
+
+        assert(virulency_rates.size() == recovery_rates.size() &&
+               virulency_rates.size() == mobility_rates.size() &&
+               recovery_rates.size() == mobility_rates.size() &&
```

Comment:
You are making redundant checks. If `virulence_rates` has the same size as recovery `recovery_rates`, and `virulence_rates` has the same size as `mobility_rates`, then mobility and recovery have the same size